File: tests/src/com.squareup.okhttp3/okhttp/3.6.0/src/test/java/android/util/Log.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package android.util;

import java.io.PrintWriter;
import java.io.StringWriter;

public final class Log {
    public static final int DEBUG = 3;
    public static final int WARN = 5;

    private Log() {
    }

    public static int println(int priority, String tag, String message) {
        return message.length();
    }

    public static String getStackTraceString(Throwable throwable) {
        StringWriter writer = new StringWriter();
        throwable.printStackTrace(new PrintWriter(writer));
        return writer.toString();
    }
}

File: tests/src/io.ktor/ktor-client-logging-jvm/3.4.3/src/test/kotlin/android/util/Log.kt
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package android.util

public object Log {
    private val loggedMessages: MutableList<Message> = mutableListOf()

    @JvmStatic
    public fun i(tag: String, message: String): Int {
        loggedMessages += Message(tag, message)
        return message.length
    }

    public fun clear(): Unit {
        loggedMessages.clear()
    }

    public fun messages(): List<Message> = loggedMessages.toList()

    public data class Message(public val tag: String, public val message: String)
}
